### PR TITLE
Fixes Non Traditional Spacing Errors during parsing.

### LINKED ISSUE #167: Warning parsing valid Cargo.toml
Hi, here is an example run https://github.com/AmbientRun/Ambient/actions/runs/5954923040/job/16152476085#step:6:13

that shows the warning:
_Warning: Error parsing Cargo.toml manifest, fallback to caching entire file: Error: Invalid TOML document: expected key-value, found comma_
because of `ambient_model = { path = "../model" , version = "0.3.0-dev" }`

Is this preventing caching from working? Does the Cargo.toml parsing need to be updated to account for this?

Thanks.

### PR DESCRIPTION
**Pull request by wyatt-herkamp**
Closes https://github.com/Swatinem/rust-cache/issues/167

This also updates the dependencies.

### CODE REVIEW

File: src/error.ts [hunk 1]
```diff
@@ -31,12 +31,12 @@ type TomlErrorOptions = ErrorOptions & {
 	ptr: number
 }
 
-function getLineColFromPtr (string: string, ptr: number): [ number, number ] {
+ function getLineColFromPtr (string: string, ptr: number): [ number, number ] {
```
Comment (cyyynthia) [inline src/error.ts]: Formatting

File: src/error.ts [hunk 2]
```diff
@@ -31,12 +31,12 @@ type TomlErrorOptions = ErrorOptions & {
...
 	let lines = string.slice(0, ptr).split(/\r\n|\n|\r/g)
 	return [ lines.length, lines.pop()!.length + 1 ]
 }
 
-function makeCodeBlock (string: string, line: number, column: number) {
+ function makeCodeBlock (string: string, line: number, column: number) {
```
Comment (cyyynthia) [inline src/error.ts]: Formatting

File: src/extract.ts [hunk 3]
```diff
@@ -81,7 +81,11 @@ export function extractValue (str: string, ptr: number, end?: string): [ TomlPri
 	let endPtr
 	if (c === '"' || c === "'") {
 		endPtr = getStringEnd(str, ptr)
-		return [ parseString(str, ptr, endPtr), endPtr + +(!!end && str[endPtr] === ',') ]
+		let parsed = parseString(str, ptr, endPtr)
+		if (!!end){
+			endPtr = skipUntil(str, endPtr, ',', end,end !== ']')
+		}
+		return [parsed, endPtr ]
```
Comment (cyyynthia) [inline src/extract.ts]: if (end) endPtr = skipUntil(str, endPtr, ',', end,end !== ']')
return [ parsed, endPtr ]

File: src/struct.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@
 
 import { parseString } from './primitive.js'
 import { extractValue } from './extract.js'
-import { type TomlPrimitive, skipComment, indexOfNewline, getStringEnd, skipVoid } from './util.js'
+import {type TomlPrimitive, skipComment, indexOfNewline, getStringEnd, skipVoid} from './util.js'
```
Comment (cyyynthia) [inline src/struct.ts]: Formatting

File: src/util.ts [hunk 5]
```diff
@@ -86,6 +86,8 @@ export function skipUntil (str: string, ptr: number, sep: string, end?: string)
 			return i + 1
 		} else if (c === end) {
 			return i
+		}else if (banNewLines && (c === '\n' || c === '\r' && str[i + 1] === '\n')) {
```
Comment (cyyynthia) [inline src/util.ts]: Formatting

File: tsconfig.json [hunk 6]
```diff
@@ -5,9 +5,10 @@
 		"@tsconfig/strictest/tsconfig.json"
 	],
 	"compilerOptions": {
+		"moduleResolution": "node16",
+		"module": "Node16",
 		"declaration": true,
 		"verbatimModuleSyntax": false,
-		"preserveConstEnums": false,
```
Comment (cyyynthia) [inline tsconfig.json:10]: This is an important thing to keep for code size and efficiency; the only const enum used in the codebase is internal and kept for readability. Inlining it is better.

File: tsconfig.json [hunk 7]
```diff
@@ -5,9 +5,10 @@
 		"@tsconfig/strictest/tsconfig.json"
 	],
 	"compilerOptions": {
+		"moduleResolution": "node16",
+		"module": "Node16",
```
Comment (cyyynthia) [inline tsconfig.json]: Aren't those already properly set by `node-lts`?

**Review by cyyynthia — CHANGES_REQUESTED** [review]
Thanks a lot for the PR, and HUGE THANKS for adding tests!!! I'm surprised this wasn't caught by the toml-test suite, it might be a good contribution to the [test suite](https://github.com/toml-lang/toml-test) as well.

I only have a few formatting nitpicks and remarks on the tsconfig changes. Great work! ❤️

File: tsconfig.json (hunk 7 above)
Comment (wyatt-herkamp) [reply tsconfig.json]: No typescript errored out for no reason?

File: tsconfig.json (hunk 6 above)
Comment (wyatt-herkamp) [reply tsconfig.json:10]: It can not be enabled while isolated modules is enabled as a note.

File: tsconfig.json (hunk 7 above)
Comment (cyyynthia) [reply tsconfig.json]: Huh, that's weird... These should be set by the node-lts preset https://github.com/tsconfig/bases/blob/main/bases/node-lts.json

Comment (wyatt-herkamp) [reply tsconfig.json]: I will revert the ts config and you can try figuring it out. Everytime I mess with the ts config I want to bang my head against a wall.

Comment (cyyynthia) [reply tsconfig.json]: I feel you! I've stopped worrying about this and use tsbases for this very reason 😄

If you need a helping hand I can help you troubleshoot, feel free to comment here with the details of the problem. 🙂

File: tsconfig.json (hunk 6 above)
Comment (cyyynthia) [reply tsconfig.json:10]: Isolated modules is not a concern here, it's safe to disable as well if it causes problems.

File: tsconfig.json (hunk 7 above)
Comment (wyatt-herkamp) [reply tsconfig.json]: After testing isolatedModules can be false and everything works.  
  
Then by putting the `"@tsconfig/node-lts/tsconfig.json"` on the list of extends it fixes the   
module Resolution Error

**Review by cyyynthia — APPROVED** [review]

**Comment by cyyynthia** [comment]
Huge thanks! I'll publish the updated version later today or tomorrow morning (CEST). ❤️ I'll post a comment here when it is published.

**Comment by cyyynthia** [comment]
This has now been published as v1.1.3. Thanks again for the contribution! ❤️

**Pull request merged**